sklearn/cluster/_affinity_propagation.py

```diff
@@ -133,7 +141,15 @@ def affinity_propagation(S, preference=None, convergence_iter=15, max_iter=200,
                     if return_n_iter
                     else (np.array([0]), np.array([0] * n_samples)))
 
-    random_state = np.random.RandomState(0)
+    if random_state == 'warn':
+        warnings.warn(("'random_state' has been introduced in 0.23. "
+                       "It will be set to None starting from 0.25 which "
+                       "means that results will differ at every function "
+                       "call. To silence this warning and obtain the same "
+                       "results as before 0.23, set it to 0."),
```

Comment:
I would suggest this to recommend users to set it explicitly to `None` rather than `0`. Maybe something like:

"Set `random_state` to `None` to silence this warning, or to `0` to keep the deprecated behavior."

Reply:
deprecated -> previous

there's no deprecated behaviour

Reply:
Sure, I don't mind. I consider the default value `0` as the random seed _deprecated_, but don't really mind.